common/lib/xmodule/xmodule/modulestore/mongo/base.py

```diff
@@ -859,7 +864,13 @@ def _cache_children(self, course_key, items, depth=0):
         course_key = self.fill_in_run(course_key)
         parent_cache = self._get_parent_cache(self.get_branch_setting())
 
-        while to_process and depth is None or depth >= 0:
+        should_process = False
```

Comment:
I think a simpler version of this would be:
```
should_process = True
if depth == 0:
    should_process = False
```

What is the error being fixed here?  It seems like there might be an issue higher up that resulted in us needing this.

Reply:
the value of depth is hardcoded as `None` (openedx/core/djangoapps/content/block_structure/factory.py:56) and python 3 doesn't allow `>=` to be performed between `None` and `int`

Reply:
we can't have `None` and `>=` check on the same level although the value is hardcoded we might need to figure out why is it hardcoded and if it's why do we have a comparison check?

Reply:
you're right it works on python shell, somehow it fails with `lms-shell`

![image](https://user-images.githubusercontent.com/19369651/63582252-2e1cca00-c5b2-11e9-92b3-0980497a57af.png)

Reply:
```
to_process and (depth is not None and depth >= 0)

Reply:
@feanil following will fail in `python` shell, I think this is the case where it fails

```
>>> depth=None
>>> False and depth is None or depth >= 0
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: unorderable types: NoneType() >= int()
```

Reply:
So the parentheses are important here, perhaps just add those?

Reply:
yeah, I added the parentheses.